widgets/gauge/gauge.go

```diff
@@ -284,6 +286,25 @@ func (g *Gauge) Draw(cvs *canvas.Canvas, meta *widgetapi.Meta) error {
 			return err
 		}
 	}
+	if g.opts.threshold > 0 && g.opts.threshold < g.total {
```

Comment:
In order to keep the Draw() function focused on higher level logic - could we move this new code into private helper methods? So that the resulting code here would look something like this:

```go
  if g.thresholdVisible() {
    if err := g.drawThreshold(...); err != nil { ... }
  }
```